### ISSUE TITLE
Provenance tracking features?

### ISSUE THREAD

**Issue by johnomotani** [boutproject/BOUT-configs#8]
One of the nice things about having a repo like this with version-controlled configurations is that it makes it easier to do provenance tracking. Just having the repo exist is a pretty good start, as e.g. STORM can grab the commit hash (plus git diff if there is one) and save it. Just wondering if we could add any features to this repo (and/or BOUT-dev) to make provenance tracking easier... For example when configuring could we package up some info about BOUT-configs (git hash + git commit + which configuration [i.e. machine] is being used + ??) and inject it into some CMake variables that BOUT could then put into `revision.hxx` or something and save to dump files. Could maybe have some shared utility scripts that could be called in different configurations to create the CMake arguments.

**Comment by ggeorgakoudis** [comment]
Is this issue relevant anymore?

**Comment by johnomotani** [comment]
> Is this issue relevant anymore?

Yes, still seems like it would be nice to have if anyone had time to implement.
